server/container/core/src/main/java/org/apache/james/server/core/MimeMessageCopyOnWriteProxy.java

```diff
@@ -92,7 +92,7 @@ public synchronized MimeMessage getWrapped() {
     protected MessageReferenceTracker refCount;
 
     public MimeMessageCopyOnWriteProxy(MimeMessage original) throws MessagingException {
-        this(new MimeMessageWrapper(original), false);
+        this(original, false);
```

Comment:
Nope fail.

I bet that:

```
new MimeMessageCopyOnWriteProxy(mimeMessage).saveChanges();
```

will update the messageID !

If for any reason the changeset I propose do not work, can you at the very least override `updateMessageId()` method to avoid such side effects.

Reply:
a false parameter means (writable = false), so I thought you would not be able to saveChanges()